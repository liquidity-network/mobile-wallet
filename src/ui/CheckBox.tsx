import React, { PureComponent } from 'react'
import { TouchableOpacity } from 'react-native'
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'

import { smallHitSlop, WHITE } from 'globalStyles'

interface Props {
  value: boolean
  onChange: (value: boolean) => void
  color?: string
  testID?: string
}

export default class CheckBox extends PureComponent<Props> {
  onChange = () => this.props.onChange(!this.props.value)

  render() {
    return (
      <TouchableOpacity
        onPress={this.onChange}
        hitSlop={smallHitSlop}
        activeOpacity={0.7}
        testID={this.props.testID}
      >
        <MaterialCommunityIcons
          name={this.props.value ? 'checkbox-marked-outline' : 'checkbox-blank-outline'}
          size={26}
          color={this.props.color || WHITE}
        />
      </TouchableOpacity>
    )
  }
}
