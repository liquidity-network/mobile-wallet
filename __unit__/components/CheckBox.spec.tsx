import React from 'react'
import { create } from 'react-test-renderer'

import CheckBox from 'ui/CheckBox'

describe('<CheckBox> component', () => {
  it('renders enabled correctly', () => {
    const tree = create(<CheckBox value onChange={() => null} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders disabled correctly', () => {
    const tree = create(<CheckBox value={false} onChange={() => null} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
