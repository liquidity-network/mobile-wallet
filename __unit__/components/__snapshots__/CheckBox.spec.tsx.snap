// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox> component renders disabled correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 7,
      "left": 7,
      "right": 7,
      "top": 7,
    }
  }
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 26,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`<CheckBox> component renders enabled correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 7,
      "left": 7,
      "right": 7,
      "top": 7,
    }
  }
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 26,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
